### ISSUE TITLE
Firebase crash due to enumerable key

### ISSUE THREAD

**Issue by shiroganee** [invertase/react-native-firebase#331]
### Environment

1. Target Platform (e.g. iOS, Android): Both
2. Development Operating System (e.g. macOS Sierra, Windows 10): macOS Sierra
3. Build tools (Xcode or Android Studio version, iOS or Android SDK version, if relevant): newest Xcode, iOS 10.3, Android 7.0 (any version is crashing)
4. React Native version (e.g. 0.45.1): 45.1+
5. RNFirebase Version (e.g. 2.0.2): 2.0.1+

Well, been trying to add a Firebase analytics to my app. After doing that wrapper where I initialize my Firebase instance:

```

import RNfirebase from 'react-native-firebase';

const instance = RNfirebase.initializeApp({
  debug: true,
  errorOnMissingPlayServices: false,
  persistence: true,
});

export default instance;
```

and later on reusing that instance in other places, i get whenever I import that file a message as 
on following screen.

<img width="371" alt="screen shot 2017-08-11 at 10 35 54" src="https://user-images.githubusercontent.com/9503398/29206138-e592ba44-7e80-11e7-84f7-08f4b37d0553.png">

I figured out that the problem lies 

`import AdMob, { statics as AdMobStatics } from './modules/admob';`

in main Firebase file which later i resolved to happen in AdMobComponent.js

Anyone had something similar? If i will manually remove the following import in Firebase.js then it start to work.

**Comment by Ehesp** [comment]
This is a new one! Would it be possible if you upgraded to the latest version of this lib for us? Makes it much easier to debug as we're constantly making updates to how the projects are built etc.

**Comment by shiroganee** [comment]
Did it, exactly same result @Ehesp.
I actually started with all newest versions. right now i'm on 46.4 RN and 2.1.1 RNfirebase.

I commented out 
`import AdMob, { statics as AdMobStatics } from './modules/admob';`
 in file firebase.js and 
`export const AdMob = require('./lib/modules/admob');`
 in index.js

**Comment by chrisbianca** [comment]
@shiroganee Can you try changing the line to:

`import AdMob from './modules/admob';`

And see if that works?

**Comment by chrisbianca** [comment]
Also, how are you using the reference in other places?

**Comment by shiroganee** [comment]
@chrisbianca  tried the one with changing the line - same result.

I reuse it later for example like that


```
import { Component, PropTypes } from 'react';
import firebase from './Init';
export default class TrackScreen extends Component {
  static propTypes = {
    screenName: PropTypes.string.isRequired,
    children: PropTypes.oneOfType([PropTypes.object, PropTypes.node, PropTypes.array])
  };

  constructor(props) {
    super();

    firebase.analytics().setCurrentScreen(props.screenName);
  }

  render() {
    return this.props.children;
  }
}

```

i'm not using AdMob at all if it is any of help

**Comment by chrisbianca** [comment]
And the same error happens on iOS and Android?
Do you have the Admob Pod added on iOS and the admob library added in your build.gradle on Android? (You shouldn't need to, I'm just curious what your setup is)

**Comment by shiroganee** [comment]
yep, exactly same error happens on both. I inserted above podfile configuration for you, something similar is set in build.gradle. And in both cases if i will remove the code that i mentioned, it wills tart to work.

```
  compile(project(':react-native-firebase')) {
        transitive = false
    }
    compile "com.google.firebase:firebase-core:11.0.4"

    // If you are receiving Google Play API availability issues, add the following dependency
    compile "com.google.android.gms:play-services-base:11.0.4"

    // RNFirebase optional dependencies
    compile "com.google.firebase:firebase-analytics:11.0.4"
    compile "com.google.firebase:firebase-crash:11.0.4"
```

**Comment by Ehesp** [comment]
@shiroganee Could you comment out these lines for me please:

https://github.com/invertase/react-native-firebase/blob/master/lib/modules/admob/index.js#L90-L93

I have a feeling because these are Classes & Components it's having a hard time assigning them. Why this only happens for you I don't know...

**Comment by Ehesp** [comment]
Out of interest, what's your Node version? Something JS specific is causing this to happen which doesn't occur on ours.

**Comment by Ehesp** [comment]
I also just pushed this: https://github.com/invertase/react-native-firebase/commit/0c69eb52377b6a453ecc02cbdbdfbc18d7d07a18

Shouldn't have been there, but maybe your setup was reading from it..!

**Comment by shiroganee** [comment]
ok so i will use git version then and give you a report back.
also i thought about Node the same thing. I used at start 6.10.2, tried 8.1.2 and result was the same.

Will give a try to these 2 other options that you mentioned. (commenting out and getting github version with newest push). Will report back.

**Comment by GabrielRivera21** [comment]
Faced same issue here using latest version of `react-native-firebase`.
commented out Admob Lines from:

- `index.js` 
- `firebase.js`

and it started to work.

### Environment

- Target Platform (e.g. iOS, Android): Both
- Development Operating System: macOS Sierra 10.12.5
- Build tools: Xcode 8.3.3, iOS 10.3, Android 6.0
- React Native version (e.g. 0.44.1): 0.43.4
- RNFirebase Version: 2.1.2
- Node v6.11.2

#### Original Stackstace:
```
<Critical>: Exception '-[NSNull integerValue]: unrecognized selector sent to instance 0x10c836180' was thrown while invoking updateExceptionMessage on target ExceptionsManager with params (
	    "One of the sources for assign has an enumerable key on the prototype chain. Are you trying to assign a prototype property? We don't allow it, as this is an edge case that we do not support. This error is a performance optimization and not spec compliant.",
	        (
	                {
	            column = 12;
	            file = "~/node_modules/react-native/packager/src/Resolver/polyfills/Object.es6.js";
	            lineNumber = 56;
	            methodName = assign;
	        },
	                {
	            column = "<null>";
	            file = "<null>";
	            lineNumber = "<null>";
	            methodName = "<unknown>";
	        },
	                {
	            column = 13;
	            file = "~/node_modules/react-native/packager/src/Resolver/polyfills/require.js";
	            lineNumber = 173;
	            methodName = loadModuleImplementation;
	        },
	                {
	            column = 0;
	            file = "~/node_modules/react-native-firebase/lib/modules/admob/Banner.js";
	            lineNumber = 2;
	            methodName = "<unknown>";
	        },
	                {
	            column = 13;
	            file = "~/node_modules/react-native/packager/src/Resolver/polyfills/require.js";
	            lineNumber = 173;
	            methodName = loadModuleImplementation;
	        },
	                {
	            column = 0;
	            file = "~/node_modules/react-native-firebase/lib/modules/admob/index.js";
	            lineNumber = 8;
	            methodName = "<unknown>";
	        },
	                {
	            column = 13;
	            file = "~/node_modules/react-native/packager/src/Resolver/polyfills/require.js";
	            lineNumber = 173;
	            methodName = loadModuleImplementation;
	        },
	                {
	            column = 29;
	            file = "~/node_modules/react-native-firebase/index.js";
	            lineNumber = 3;
	            methodName = "<unknown>";
	        },
	                {
	            column = 13;
	            file = "~/node_modules/react-native/packager/src/Resolver/polyfills/require.js";
	            lineNumber = 173;
	            methodName = loadModuleImplementation;
	        },
	                {
	            column = 0;
	            file = "~/build/firebase/index.js";
	            lineNumber = 1;
	            methodName = "<unknown>";
	        },
```

I'll see if I can get an MCVE for you guys.

**Comment by Ehesp** [comment]
Thanks @GabrielRivera21 - out of interest could you just try bumping your node version to a later version for us?

**Comment by GabrielRivera21** [comment]
Used 8.4.0 and same issue.

**Comment by GabrielRivera21** [comment]
Once I can replicate the behaviour in an MCVE, we'll be able to identify the root cause, but unfortunately, I won't have time to make it until the weekend.

**Comment by Ehesp** [comment]
Crazy! An MVP would be awesome, I cannot replicate this myself and tried on a couple of environments.

**Comment by Salakar** [comment]
@GabrielRivera21 did you try locally change your version to match the change @Ehesp pushed up:  https://github.com/invertase/react-native-firebase/commit/0c69eb52377b6a453ecc02cbdbdfbc18d7d07a18

Wondering if it's that

**Comment by GabrielRivera21** [comment]
It matches.

**Comment by blrichards** [comment]
I started pull request #348 to attempt to fix this

**Comment by GabrielRivera21** [comment]
Sorry I havent had time to do an MCVE but I'll test out your PR to see if the issue fixes itself

**Comment by blrichards** [comment]
Sounds good, fixed the issue for me

**Comment by GabrielRivera21** [comment]
PR #348 fixed it for me. Thanks @blrichards

**Comment by Ehesp** [comment]
Merged it... Will get a release up today.

**Comment by chrisbianca** [comment]
I've just released v2.1.3 which includes this fix
